repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py

```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
         filter = TagFilter(ReportPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class AttachPackageReposPhase(Phase):
+        '''Make sure the repositories of the new system are available.'''
         name = 'AttachPackageRepos'
         filter = TagFilter(AttachPackageReposPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class PlanningPhase(Phase):
+        '''RPM Install transaction (feasibility checks).'''
         name = 'Planning'
         filter = TagFilter(PlanningPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class DownloadPhase(Phase):
+        '''Download of all things needed (RPMs, Kernel images, Anaconda image
+        (very big maybe), etc.)
+        '''
         name = 'Download'
         filter = TagFilter(DownloadPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class InterimPreparationPhase(Phase):
+        '''Preparation of initial ramdisk (if required).
+        Setup bootloaderGrub.
+        '''
         name = 'InterimPreparation'
         filter = TagFilter(InterimPreparationPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags(request_restart_after_phase=True)
 
     class InitRamStartPhase(Phase):
+        '''Reboot into the next system therefore not a real step, however the
+        before and post parts are interesting to have for hooking.
+        '''
         name = 'InitRamStart'
         filter = TagFilter(InitRamStartPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class NetworkPhase(Phase):
+        '''Bring up the network.'''
         name = 'Network'
         filter = TagFilter(NetworkPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class StoragePhase(Phase):
+        '''Mount all storage points.'''
         name = 'Storage'
         filter = TagFilter(StoragePhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class LateTestsPhase(Phase):
+        '''Tests that need to be executed later.'''
         name = 'LateTests'
         filter = TagFilter(LateTestsPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class PreparationPhase(Phase):
+        '''YUM Configurations. Backup not upgradable configurations (No Verify
+        Configs).
+        '''
```

Comment:
No Verify Configs != not upgradable configuration

Reply:
I think that this should be called PreTransaction and should consider to apply workarounds etc that are needed to make the transaction successful for example.